src/components/header/gh-version.jsx

```diff
@@ -1,24 +1,22 @@
 import { useEffect, useState } from 'preact/hooks';
-import { fetchRelease } from '../../lib/github';
+import { fallbackData, fetchRelease } from '../../lib/github';
 import config from '../../config.json';
 
-const releaseURL = version =>
-	`https://github.com/preactjs/preact/releases/tag/${version}`;
-
-export default function ReleaseLink({ preactVersion, ...props }) {
-	const [url, setUrl] = useState(releaseURL(preactVersion));
-	const [version, setVersion] = useState(preactVersion);
+export default function ReleaseLink({ ...props }) {
+	const [release, setRelease] = useState({
+		url: globalThis.prerenderPreactReleaseUrl || fallbackData.preactReleaseUrl,
+		version: globalThis.prerenderPreactVersion || fallbackData.preactVersion
+	});
 
 	useEffect(() => {
 		fetchRelease(config.repo).then(d => {
-			setVersion(d.version);
-			setUrl(d.url);
+			setRelease(d);
 		});
```

Comment:
This is a bit odd to me and I'm not quite sure if we have a bug or if it's just late and I'm being daft.

Assume we prerender for v10.19.4, generating the following HTML-ish:

```html
<a href="https://github.com/preactjs/preact/releases/tag/10.19.4">v10.19.4</a>
```

Upon hydration, say the fetch fails but catches and returns the info for `10.10.0`.

The text child of the link will be updated to `v10.10.0` as you'd expect, but the `href` will remain the same (`.../releases/tag/10.19.4`). However, if you set the URL to something wildly different before updating the state value (`d.url = 'https://google.com'; setRelease(d);`), this will update the `href`.

Are we doing something clever to try to avoid updating the attr unnecessarily?